Framework/DataHandling/src/FindDetectorsPar.cpp

```diff
@@ -625,7 +625,7 @@ FindDetectorsPar::get_ASCII_header(std::string const &fileName,
     file_descriptor.data_start_position =
         data_stream.tellg(); // if it is PHX or PAR file then the data begin
                              // after the first line;
-    file_descriptor.nData_records = atoi(&BUF[0]);
+    file_descriptor.nData_records = std::stoi(&BUF[0]);
```

Comment:
While you're modernizing, change `&BUF[0]` to [`BUF.data()`](http://en.cppreference.com/w/cpp/container/vector/data).